sharing/filter_range_values.py

```diff
@@ -0,0 +1,107 @@
+"""
+This module filters the data by a range between two datetime values.
+"""
+
+from typing import Tuple, Union  # pylint: disable=import-error
+from datetime import datetime, timedelta  # pylint: disable=import-error
+import re
+import pandas as pd  # pylint: disable=import-error
+from pandas.core.frame import DataFrame  # pylint: disable=import-error
+
+
+def filter_range_value(
+    column: bool,
+    bracket: str,
+    low_limit: str,
+    up_limit: str,
+    intermediate_filtered_data: DataFrame,
+    var: str,
+) -> Tuple[DataFrame, bool, bool, bool]:
+    """The function filters data based on the type of range interval.
+
+    Function returns back the dataframe with or without filter operation.
+
+    Parameter:
+        column(bool): checks whether the rule direction is column
+        bracket(str): type of range bracket
+        low_limit(str): lower limit of the range
+        up_limit(str): upper limit of the range
+        intermediate_filtered_data(dataframe): data to be filtered
+        var(str): variable to be filtered by from current rule
+    Returns:
+        dataframe: Dataframe that is filtered if rule value is of datetime type
+    """
+
+    if not column:
```

Comment:
All of this logic should really be wrapped up into an Interval class to avoid duplicating a lot of the logic in this function. 

The fields in this class are:
lowerLimit: number | datetime
upperLimit: number | datetime
isLeftOpen: bool
isRightOpen: bool

Class Methods:
areValuesWithinInterval(values: List[Union[number, datetime]]): boolean